backup/backup_test.go

```diff
@@ -110,30 +193,30 @@ func TestChoosePod(t *testing.T) {
 		bkup     mocov1beta2.BackupStatus
 		pods     []*corev1.Pod
 
-		expectIdx int
+		expectIdx        int
+		skipBackupBinlog bool
+		warnings         int
 	}{
-		{"single", 1, 0, mocov1beta2.BackupStatus{}, makePod1(true), 0},
-		{"single-not-ready", 1, 0, mocov1beta2.BackupStatus{}, makePod1(false), 0},
-		{"triple-ready", 3, 0, mocov1beta2.BackupStatus{}, makePod3(true, false, true), 2},
-		{"triple-not-ready", 3, 1, mocov1beta2.BackupStatus{}, makePod3(false, true, false), 1},
-		{"single-2nd", 1, 0, makeBS(0, "123"), makePod1(true), 0},
-		{"single-2nd-uuid-changed", 1, 0, makeBS(0, "abc"), makePod1(true), 0},
-		{"single-2nd-not-ready", 1, 0, makeBS(0, "123"), makePod1(false), 0},
-		{"triple-2nd", 3, 0, makeBS(1, "123"), makePod3(true, true, true), 1},
-		{"triple-2nd-uuid-changed", 3, 0, makeBS(1, "abc"), makePod3(true, true, true), 2},
-		{"triple-2nd-not-ready", 3, 0, makeBS(1, "123"), makePod3(true, false, true), 2},
-		{"triple-2nd-primary", 3, 1, makeBS(1, "123"), makePod3(true, true, true), 0},
-		{"triple-2nd-all-not-ready", 3, 0, makeBS(1, "123"), makePod3(true, false, false), 0},
+		{"single", 1, 0, mocov1beta2.BackupStatus{}, makePod1(true), 0, true, 0},
+		{"single-not-ready", 1, 0, mocov1beta2.BackupStatus{}, makePod1(false), 0, true, 0},
+		{"triple-ready", 3, 0, mocov1beta2.BackupStatus{}, makePod3(true, false, true), 2, true, 0},
+		{"triple-not-ready", 3, 1, mocov1beta2.BackupStatus{}, makePod3(false, true, false), 1, true, 0},
+		{"single-2nd", 1, 0, makeBS(1, 0, "uuid-0", map[string]string{"0": "uuid-0"}), makePod1(true), 0, false, 0},
+		{"single-2nd-uuid-changed", 1, 0, makeBS(1, 0, "uuid-a", map[string]string{"0": "uuid-a"}), makePod1(true), 0, true, 1},
+		{"single-2nd-not-ready", 1, 0, makeBS(1, 0, "uuid-0", map[string]string{"0": "uuid-0"}), makePod1(false), 0, true, 1},
+		{"triple-2nd", 3, 0, makeBS(3, 1, "uuid-1", map[string]string{"0": "uuid-0", "1": "uuid-1", "2": "uuid-2"}), makePod3(true, true, true), 1, false, 0},
+		{"triple-2nd-uuid-changed", 3, 0, makeBS(3, 1, "uuid-b", map[string]string{"0": "uuid-0", "1": "uuid-b", "2": "uuid-2"}), makePod3(true, true, true), 2, false, 0},
+		{"triple-2nd-not-ready", 3, 0, makeBS(3, 1, "uuid-1", map[string]string{"0": "uuid-0", "1": "uuid-1", "2": "uuid-2"}), makePod3(true, false, true), 2, false, 0},
+		{"triple-2nd-primary", 3, 1, makeBS(3, 1, "uuid-1", map[string]string{"0": "uuid-0", "1": "uuid-1", "2": "uuid-2"}), makePod3(true, true, true), 0, false, 0},
+		{"triple-2nd-all-not-ready", 3, 0, makeBS(3, 1, "uuid-1", map[string]string{"0": "uuid-0", "1": "uuid-1", "2": "uuid-2"}), makePod3(true, false, false), 0, false, 0},
```

Comment:
I think it would be nice to add test case showing the case where the uuid of all replicas are changed like this.
```
{"triple-2nd-replica-uuid-changed", 3, 0, makeBS(3, 1, "uuid-a", map[string]string{"0": "uuid-0", "1": "uuid-a", "2": "uuid-b"}), makePod3(true, true, true), 0, false, 0},
```